### PR TITLE
[connectedvmware] Enable VM using moName

### PR DESCRIPTION
**Pull request by nascarsayan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

```yaml
$ az connectedvmware vm create -g contoso-rg -v contoso-vc --mo-name contoso-vm -n test-mo

2024-09-16T00:56:31 WARNING  [1] /subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/vm-123456 
{
  "inventoryType": "VirtualMachine",
  "provisioningState": "Succeeded",
  "managedResourceId": "",
  "moName": "contoso-vm",
  "toolsVersionStatus": "guestToolsSupportedOld",
  "toolsRunningStatus": "guestToolsRunning",
  "toolsVersion": "11297",
  "instanceUuid": "12345678-1234-1234-1234-123456789012",
  "resourcePool": {
    "inventoryType": "ResourcePool",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/resgroup-123456",
    "moName": "contoso-rp"
  },
  "ipAddresses": [
    "fe80::1234:5678:9abc:def0",
    "192.168.0.1"
  ],
  "powerState": "poweredOn",
  "smbiosUuid": "12345678-1234-1234-1234-123456789012",
  "folderPath": "Contoso-Datacenter",
  "moRefId": "vm-123456",
  "osType": "Windows",
  "osName": "Microsoft Windows Server 2019 (64-bit)",
  "host": {
    "inventoryType": "Host",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/host-123456",
    "moName": "contoso-host1"
  },
  "cluster": {
    "inventoryType": "Cluster",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/domain-1234",
    "moName": "Cluster-1"
  },
  "computerName": "contoso-vm"
}
2024-09-16T00:56:31 WARNING  [2] /subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/vm-123457 
{
  "inventoryType": "VirtualMachine",
  "provisioningState": "Succeeded",
  "managedResourceId": "",
  "moName": "contoso-vm",
  "toolsVersionStatus": "guestToolsSupportedOld",
  "toolsRunningStatus": "guestToolsRunning",
  "toolsVersion": "11297",
  "instanceUuid": "12345678-1234-1234-1234-123456789013",
  "resourcePool": {
    "inventoryType": "ResourcePool",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/resgroup-123456",
    "moName": "contoso-rp"
  },
  "ipAddresses": [
    "fe80::1234:5678:9abc:def1",
    "192.168.0.2"
  ],
  "powerState": "poweredOn",
  "smbiosUuid": "12345678-1234-1234-1234-123456789013",
  "folderPath": "Contoso-Datacenter",
  "moRefId": "vm-123457",
  "osType": "Windows",
  "osName": "Microsoft Windows Server 2019 (64-bit)",
  "host": {
    "inventoryType": "Host",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/host-123457",
    "moName": "contoso-host2"
  },
  "cluster": {
    "inventoryType": "Cluster",
    "inventoryItemId": "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/domain-1234",
    "moName": "Cluster-1"
  },
  "computerName": "contoso-vm-1"
}
2024-09-16T00:56:31 WARNING  Multiple inventory items found with mo_name 'contoso-vm' in vCenter 'contoso-vc'. Please check logs above for the properties of the inventory items.
Do you want to select one of the inventory items to proceed? (y/n): y
2024-09-16T00:56:35 WARNING  [1] /subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/vm-123456
2024-09-16T00:56:35 WARNING  [2] /subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/vm-123457
Select one of the inventory items by entering the index between 1 and 2: 12
Invalid index. Please enter a valid index between 1 and 2: 2
2024-09-16T00:56:39 WARNING  You selected: /subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/contoso-rg/providers/Microsoft.ConnectedVMwarevSphere/VCenters/contoso-vc/InventoryItems/vm-123457
```

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedvmware/azext_connectedvmware/_params.py [hunk 1]
```diff
@@ -37,6 +39,11 @@ def load_arguments(self, _):
         help='Name or ID of the inventory item.',
     )
 
+    mo_name = CLIArgumentType(
+        options_list=['--mo-name'],
+        help='Name of the managed object.',
```
Comment (subbartt) [inline src/connectedvmware/azext_connectedvmware/_params.py:44]: Name of the resource in the VCenter.

File: src/connectedvmware/azext_connectedvmware/custom.py [hunk 2]
```diff
@@ -987,6 +994,103 @@ def create_vm(
...
+                            "You selected: %s",
+                            vm_list[selected - 1]["id"]
+                        )
+                        break
+                    selected = prompt(
+                        f"Invalid index. Please enter a valid index between 1 and {len(vm_list)}: "
+                    )
+            except ValueError:
+                raise InvalidArgumentValueError(
+                    "Invalid input. Please provide a valid index."
```
Comment (subbartt) [inline src/connectedvmware/azext_connectedvmware/custom.py:1088]: Let us validate this code path.

**Review by subbartt — APPROVED** [review]

File: src/connectedvmware/azext_connectedvmware/_params.py (hunk 1 above)
Comment (nascarsayan) [reply src/connectedvmware/azext_connectedvmware/_params.py]: updated

File: src/connectedvmware/azext_connectedvmware/custom.py (hunk 2 above)
Comment (nascarsayan) [reply src/connectedvmware/azext_connectedvmware/custom.py:1088]: Have done the validation, and added the logs in the PR description.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**